File: ai/app/schemas/action_type.py
from __future__ import annotations

from enum import Enum


class ActionType(str, Enum):
    IMPRESSION = "IMPRESSION"
    LIKE = "LIKE"
    DISLIKE = "DISLIKE"
    BOOKMARK = "BOOKMARK"
    VIEW_DETAIL = "VIEW_DETAIL"
    VIEW_SPECIFICATION = "VIEW_SPECIFICATION"

    @property
    def weight(self) -> float:
        return {
            ActionType.IMPRESSION: 0.1,
            ActionType.LIKE: 1.0,
            ActionType.DISLIKE: -1.0,
            ActionType.BOOKMARK: 0.7,
            ActionType.VIEW_DETAIL: 0.3,
            ActionType.VIEW_SPECIFICATION: 0.5,
        }[self]

    @classmethod
    def preference_initializable(cls) -> tuple["ActionType", ...]:
        return (
            cls.LIKE,
            cls.DISLIKE,
            cls.BOOKMARK,
            cls.VIEW_DETAIL,
            cls.VIEW_SPECIFICATION,
        )

    @classmethod
    def preference_initializable_values(cls) -> list[str]:
        return [action_type.value for action_type in cls.preference_initializable()]

File: backend/src/main/java/site/to_mato/recommendation/entity/enums/ActionType.java
package site.to_mato.recommendation.entity.enums;

public enum ActionType {
    IMPRESSION,         // 추천 목록에 노출
    VIEW_DETAIL,        // 상세 조회
    VIEW_SPECIFICATION, // 구체화 조회

    LIKE,               // 좋아요
    LIKE_CANCEL,

    DISLIKE,            // 싫어요
    DISLIKE_CANCEL,

    BOOKMARK,           // 북마크
    BOOKMARK_CANCEL
}
